connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -111,6 +111,23 @@ public RetryWithToleranceOperator(long errorRetryTimeout, long errorMaxDelayInMi
         return errantRecordFuture;
     }
 
+    public synchronized Future<Void> executeFailed(Stage stage, Class<?> executingClass,
+                                                   SourceRecord sourceRecord,
+                                                   Throwable error) {
+
+        markAsFailed();
+        context.sourceRecord(sourceRecord);
+        context.currentContext(stage, executingClass);
+        context.error(error);
+        errorHandlingMetrics.recordFailure();
+        Future<Void> errantRecordFuture = context.report();
+        if (!withinToleranceLimits()) {
+            errorHandlingMetrics.recordError();
+            throw new ConnectException("Tolerance exceeded in error handler", error);
```

Comment:
Now that this message can come from 2 different paths, should we add some context to the message to disambiguate them?

Reply:
I added some context to the string error message denoting it was a Source Worker. I am open to suggestions on how verbose this message should be.